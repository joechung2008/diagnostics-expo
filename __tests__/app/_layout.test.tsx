import RootLayout from '@/app/_layout';
import { render } from '@testing-library/react-native';
import React from 'react';

// Mock providers with spies
let mockAppProvider: jest.Mock;
let mockThemeProvider: jest.Mock;
let mockSafeAreaProvider: jest.Mock;
let mockSlot: jest.Mock;

jest.mock('@/components/AppContext', () => {
  mockAppProvider = jest.fn(
    ({ children }: { children: React.ReactNode }) => children
  );
  return {
    AppProvider: mockAppProvider,
  };
});

jest.mock('@/components/ThemeContext', () => {
  mockThemeProvider = jest.fn(
    ({ children }: { children: React.ReactNode }) => children
  );
  return {
    ThemeProvider: mockThemeProvider,
  };
});

jest.mock('react-native-safe-area-context', () => {
  mockSafeAreaProvider = jest.fn(
    ({ children }: { children: React.ReactNode }) => children
  );
  return {
    SafeAreaProvider: mockSafeAreaProvider,
  };
});

jest.mock('expo-router', () => {
  mockSlot = jest.fn(() => null);
  return {
    Slot: mockSlot,
  };
});

describe('RootLayout', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders with proper provider hierarchy', () => {
    const { toJSON } = render(<RootLayout />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders all required providers and components', () => {
    render(<RootLayout />);

    expect(mockSafeAreaProvider).toHaveBeenCalledTimes(1);
    expect(mockThemeProvider).toHaveBeenCalledTimes(1);
    expect(mockAppProvider).toHaveBeenCalledTimes(1);
    expect(mockSlot).toHaveBeenCalledTimes(1);
  });

  it('renders without crashing', () => {
    expect(() => render(<RootLayout />)).not.toThrow();
  });
});
