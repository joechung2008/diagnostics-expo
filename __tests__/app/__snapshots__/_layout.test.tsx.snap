// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout renders with proper provider hierarchy 1`] = `null`;
